src/fsharp/symbols/Symbols.fs

```diff
@@ -34,7 +34,7 @@ type FSharpAccessibility(a:Accessibility, ?isProtected) =
         | _ when List.forall isInternalCompPath p  -> Internal 
         | _ -> Private
 
-    member __.IsPublic = not isProtected && match a with Public -> true | _ -> false
+    member __.IsPublic = not isProtected && match a with TAccess [] -> true | _ -> false
```

Comment:
if I recall correctly, you can mark the `Public|Internal|Private` AP `internal` and the allocation gets erased entirely so you can continue to use the AP in the idiomatic way.

Reply:
The AP is type-private, isn't it the same for optimizer?

Reply:
Example [sharplab](https://sharplab.io/#v2:DYLgZgzgPgLgngBwKYAIAqBBAxlpEIoC862u+KA9mJQEYBWKwAlhDALABQnwSMKTAO2YDUACigAFAK41mWKAEkBMJACcBAQ2CTVTAG4aVUAJQpRmHHgIJTxTigcoAtoawALFAhQB3JjA/2jlAoANoAuigAtAB8KNKyTFgogQ7BAPo+bkgCKAAyLDAAdGAUqlrAZmBSORkxKDCqUkimXlGxSirqWskcjijpbXG6BiopPX3cvCiqeFLAfHa9ji4w7mYWZAThpr7+Y8HxcoMNTfsotbFgWhBIQA)

Reply:
Did you mean `inline` instead of `internal`? :)

Reply:
Not sure about the type-private case you mention, but if you remove the `inline` in that sharplab you see the additional choice allocations occurring. It's also a very simple example and I'm sure there's additional context I'm missing.  And yes, I do mean `inline` :D

Reply:
Did more experiments and after checking the code in class members inlining results in worse code. Ignore everything I said :)